arraycontext/container/__init__.py

```diff
@@ -172,7 +172,16 @@ def _serialize_ndarray_container(ary: np.ndarray) -> Iterable[Tuple[Any, Any]]:
         raise ValueError(
                 f"only object arrays are supported, given dtype '{ary.dtype}'")
 
-    return np.ndenumerate(ary)
+    # special-cased for speed
+    if ary.ndim == 1:
+        return [(i, ary[i]) for i in range(ary.shape[0])]
```

Comment:
Shouldn't the key be a tuple
```python
In [4]: list(np.ndenumerate(np.random.rand(3)))
Out[4]: 
[((0,), 0.09292195880848719),
 ((1,), 0.6780962855552635),
 ((2,), 0.3681587065690297)]
```

Reply:
I.e.:
```python
In [6]: ary = np.random.rand(3)

In [7]: list(np.ndenumerate(ary))
Out[7]: 
[((0,), 0.646399478266698),
 ((1,), 0.015079989468580912),
 ((2,), 0.3586538930864147)]

In [8]: [(i, ary[i]) for i in range(ary.shape[0])]
Out[8]: [(0, 0.646399478266698), (1, 0.015079989468580912), (2, 0.3586538930864147)]
```

Was this intentional or a typo?

Reply:
Sorry... it merged before I had a chance to see your comment. IMO it's OK. The types of the keys are defined by the individual object, the array container widgetry doesn't care.

And for the object array deserialization, all that matters is that the key works as an index for an array of the same shape, which it does:

https://github.com/inducer/arraycontext/blob/ef662bec5f829e2586c328127d259a3347c54a86/arraycontext/container/__init__.py#L196-L197